blackjax/mcmc/integrators.py

```diff
@@ -115,61 +238,25 @@ def mclachlan(
     and derives different values.
 
     """
-    b1 = 0.1932
+    b1 = 0.1931833275037836
     a1 = 0.5
     b2 = 1 - 2 * b1
-
-    logdensity_and_grad_fn = jax.value_and_grad(logdensity_fn)
-    kinetic_energy_grad_fn = jax.grad(kinetic_energy_fn)
-
-    def one_step(state: IntegratorState, step_size: float) -> IntegratorState:
-        position, momentum, _, logdensity_grad = state
-
-        momentum = jax.tree_util.tree_map(
-            lambda momentum, logdensity_grad: momentum
-            + b1 * step_size * logdensity_grad,
-            momentum,
-            logdensity_grad,
-        )
-
-        kinetic_grad = kinetic_energy_grad_fn(momentum)
-        position = jax.tree_util.tree_map(
-            lambda position, kinetic_grad: position + a1 * step_size * kinetic_grad,
-            position,
-            kinetic_grad,
-        )
-
-        _, logdensity_grad = logdensity_and_grad_fn(position)
-        momentum = jax.tree_util.tree_map(
-            lambda momentum, logdensity_grad: momentum
-            + b2 * step_size * logdensity_grad,
-            momentum,
-            logdensity_grad,
-        )
-
-        kinetic_grad = kinetic_energy_grad_fn(momentum)
-        position = jax.tree_util.tree_map(
-            lambda position, kinetic_grad: position + a1 * step_size * kinetic_grad,
-            position,
-            kinetic_grad,
-        )
-
-        logdensity, logdensity_grad = logdensity_and_grad_fn(position)
-        momentum = jax.tree_util.tree_map(
-            lambda momentum, logdensity_grad: momentum
-            + b1 * step_size * logdensity_grad,
-            momentum,
-            logdensity_grad,
-        )
-
-        return IntegratorState(position, momentum, logdensity, logdensity_grad)
+    cofficients = [b1, a1, b2, a1, b1]
+    position_update_fn = euclidean_position_update_fn(logdensity_fn)
+    momentum_update_fn = euclidean_momentum_update_fn(kinetic_energy_fn)
+    one_step = generalized_symplectic_integrator(
+        momentum_update_fn,
+        position_update_fn,
+        cofficients,
+        format_output_fn=format_euclidean_state_output,
+    )
 
     return one_step
 
 
 def yoshida(
```

Comment:
Another minor point, but it seems like a bit of duplicated code between velocity_verlet, mclachlan and yoshida could be removed, since they are all identical except for the choice of coefficients. Why not have just one function `make_euclidean_integrator`, and then set e.g. `velocity_verlet = make_euclidean_integrator([0.5,1,05])`

Reply:
Yes good point.